clients/src/main/java/org/apache/kafka/common/protocol/types/Type.java

```diff
@@ -59,6 +59,47 @@ public boolean isNullable() {
         return false;
     }
 
+    /**
+     * The Boolean type represents a boolean value in a byte by using
+     * the value of 0 to represent false, and 1 to represent true.
+     *
+     * If for some reason a value that is not 0 or 1 is read,
+     * then any non-zero value will return true.
```

Comment:
I implemented it this way to match the existing behavior of StopReplica requests delete_partitions boolean. That way this type could be used there as well without a version bump, and all the "booleans" would behave the same. I planned to update StopReplicaRequest after this patch based on the decisions here. 

Here is the relevant code: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/requests/StopReplicaRequest.java#L83